svf/lib/AE/Svfexe/AbstractInterpretation.cpp

```diff
@@ -573,20 +574,30 @@ void AbstractInterpretation::handleWTOComponents(const std::list<const ICFGWTOCo
 {
     for (const ICFGWTOComp* wtoNode : wtoComps)
     {
-        if (const ICFGSingletonWTO* node = SVFUtil::dyn_cast<ICFGSingletonWTO>(wtoNode))
-        {
-            handleSingletonWTO(node);
-        }
-        // Handle WTO cycles
-        else if (const ICFGCycleWTO* cycle = SVFUtil::dyn_cast<ICFGCycleWTO>(wtoNode))
-        {
-            handleCycleWTO(cycle);
-        }
-        // Assert false for unknown WTO types
-        else
-        {
-            assert(false && "unknown WTO type!");
-        }
+        handleWTOComponent(wtoNode);
+    }
+}
+
+void AbstractInterpretation::handleWTOComponent(const SVF::ICFGWTOComp* wtoNode)
+{
+    if (!mergeStatesFromPredecessors(wtoNode))
+    {
+        // No ES on the in edges - Infeasible block
+        return;
+    }
+    if (const ICFGSingletonWTO* node = SVFUtil::dyn_cast<ICFGSingletonWTO>(wtoNode))
+    {
```

Comment:
if(mergeStatesFromPredecessors(wtonode->getNode()))
   handleSingletonWTO(node);

Reply:
Also rename the following:

ICFGSingletonWTO::getNode=> ICFGSingletonWTO::getICFGNode();
ICFGCycleWTO::getNode=> ICFGCycleWTO::getICFGNode();